### PR TITLE
Users/prachisingla/add monitor iterations in base monitor

### PR DESCRIPTION
**Pull request by psingla1210**

### CODE REVIEW

File: src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs [hunk 1]
```diff
@@ -14,10 +14,11 @@ namespace VirtualClient.Contracts
     public abstract class VirtualClientIntervalBasedMonitor : VirtualClientComponent
     {
         /// <summary>
-        /// The default frequency that monitors run at (every one minute)
+        /// The default frequency that monitors run at (every five minutes)
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs:17]: Thank you!

File: src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs [hunk 2]
```diff
@@ -14,10 +14,11 @@ namespace VirtualClient.Contracts
...
         /// </summary>
         private static TimeSpan defaultFrequency = TimeSpan.FromMinutes(5);
         private static TimeSpan defaultWarmupPeriod = TimeSpan.FromMinutes(5);
+        private static long defaultIterations = 6315489358112;
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs:21]: You could use something like long.maxvalue

File: src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs [hunk 3]
```diff
@@ -68,5 +69,17 @@ public TimeSpan MonitorWarmupPeriod
                     nameof(VirtualClientIntervalBasedMonitor.MonitorWarmupPeriod), VirtualClientIntervalBasedMonitor.defaultWarmupPeriod);
             }
         }
+
+        /// <summary>
+        /// The number of iterations with which this monitor runs
+        /// </summary>
+        public long MonitorIterations
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs:76]: What's the reason to put a maximum when user doesn't define it? User might want infinite monitor iterations, like SAP

**Review by yangpanMS — APPROVED** [review]

File: src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs (hunk 3 above)
Comment (psingla1210) [reply src/VirtualClient/VirtualClient.Contracts/VirtualClientIntervalBasedMonitor.cs:76]: I have set default value, reason behind it is if let say a monitor is using this to determine number of iterations and user removed the line which is providing this value in profile file then to protect the code from crashing, i gave a very high value which is like infinite only, do you see any concern?

**Review by yangpanMS — APPROVED** [review]

**Pull request merged**